workspace_tools/tests.py

```diff
@@ -1119,13 +1119,14 @@ def __init__(self, n):
         self.__dict__.update(TESTS[n])
 
     def is_supported(self, target, toolchain):
-        if hasattr(self, 'mcu') and not target in self.mcu:
-            return False
-        if hasattr(self, 'exclude_mcu') and target in self.exclude_mcu:
-            return False
-        if not hasattr(self, 'supported'):
-            return True
-        return (target in self.supported) and (toolchain in self.supported[target])
+        return True
```

Comment:
This info is still pretty useful, even if we don't necessarily want to use it at times.

I'd suggest adding an option in `singletest.py` to ignore the `is_supported` function here, but still keep the ability to use this info.

I'd suggest adding the `--include-unsupported` option to `singletest.py`. That way we keep backward compatibility but we add the ability to build all tests. We'd simply just have to check for this option here: https://github.com/mbedmicro/mbed/blob/1f340445d5333b485923beb9a7216fc54a8efeb4/workspace_tools/test_api.py#L652

If this sounds like a good option, then I'd suggest reverting the changes here (uncomment the commented code), merge the PR, and I can submit a change to `singletest.py`